### PR TITLE
Support to move IIS to GA

### PR DESCRIPTION
**Pull request by narph**
## What does this PR do?

Maps fields
Sets module and metricsets to GA

## Why is it important?

Sets module and metricsets to GA

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes https://github.com/elastic/beats/issues/22956 https://github.com/elastic/beats/issues/22609

### CODE REVIEW

File: x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml [hunk 1]
```diff
@@ -1,10 +1,76 @@
...
     - name: name
       type: keyword
       description: >
         application pool name
+    - name: process
+      type: group
+      description: >
+        process details
+      fields:
+        - name: pid
```
Comment (jsoriano) [inline x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml:16]: Is this equivalent to the pid of the operating system? If it is, this should be probably stored in the ECS `process.pid` field, and better to do the change before going GA :slightly_smiling_face: 
If it isn't the same, we could add something about this in the description.

File: x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml [hunk 2]
```diff
@@ -1,10 +1,76 @@
...
+          type: long
+          description: >
+             process id
+        - name: handle_count
+          type: long
+          description: >
+            handle count
```
Comment (jsoriano) [inline x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml:23]: These descriptions are intended for documentation, please use full sentences, e.g:
```suggestion
            Number of handles in use by the server.
```

File: x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml [hunk 3]
```diff
@@ -1,10 +1,76 @@
...
+        - name: throw_to_catch_depth_per_sec
+          type: long
+          description: >
+            throw to catch depth per sec
+        - name: total_exceptions_thrown
+          type: long
+          description: >
+            total exceptions thrown
+        - name: filters_per_sec
+          type: long
```
Comment (jsoriano) [inline x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml:74]: Should these `...per_sec` metrics be defined as float?

File: x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml (hunk 1 above)
Comment (narph) [reply x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml]: the field is equivalent to the pid and has been mapped as such in the code, I have removed the mapping inside the fields.yml file.

File: x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml (hunk 3 above)
Comment (narph) [reply x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml]: updated to them to `float`

File: x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml [hunk 4]
```diff
@@ -1,10 +1,72 @@
...
+            Memory page faults.
+    - name: net_clr
+      type: group
+      description: >
+        net clr
+      fields:
+        - name: finallys_per_sec
+          type: float
+          description: >
+            The nr of finallys per sec.
```
Comment (jsoriano) [inline x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml:60]: Nit.
```suggestion
            The number of finallys per sec.
```

File: x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml [hunk 5]
```diff
@@ -1,10 +1,72 @@
...
+          description: >
+            Memory private bytes.
+        - name: page_faults_per_sec
+          type: float
+          description: >
+            Memory page faults.
+    - name: net_clr
+      type: group
+      description: >
+        net clr
```
Comment (jsoriano) [inline x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml:55]: There is an obscure description still here.

File: x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml [hunk 6]
```diff
@@ -1,10 +1,72 @@
 - name: application_pool
   type: group
-  release: beta
+  release: ga
   description: >
     application_pool
```
Comment (jsoriano) [inline x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml:5]: Improve this description.

**Review by jsoriano — APPROVED** [review]
Some nitpicking comments in some descriptions, but LGTM. Thanks!

File: x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml (hunk 4 above)
Comment (narph) [reply x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml]: fixed

File: x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml (hunk 5 above)
Comment (narph) [reply x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml]: just added description to all the group fields, I missed those fields somehow.

File: x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml (hunk 6 above)
Comment (narph) [reply x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml]: same here

File: x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml (hunk 5 above)
Comment (jsoriano) [reply x-pack/metricbeat/module/iis/application_pool/_meta/fields.yml]: :+1:

**Review by jsoriano — APPROVED** [review]
:+1:

**Pull request merged**